[ios][extensions]fix memory leak, and add instruction for advanced scrolling behavior (#11882)
flutter/website: src/content/platform-integration/ios/app-extensions.md

   [Adding a Flutter Screen][]. For example, you can display a
   specific route in your Flutter app within a share extension.

> 1. Cancel the request in `viewDidDisappear`, which will be called when the share extension is dismissed by gesture. 

Reviewer: I don't see a "share" extension. Does you mean share_plus?
If so, we should say `share_plus`.

```suggestion
1. Cancel the request in `viewDidDisappear`,
    which is called when the `share` extension is dismissed with a gesture. 
```

Author: I mean "share extension" which is one kind of app extensions. We use "share extension" as the example in this section: https://flutter-docs-prod--pr11882-share-extension-memory-leak-73qwnwz8.web.app/platform-integration/ios/app-extensions#creating-app-extension-uis-with-flutter

Reviewer: OK! I didn't understand what a "share extension" was. Now that I do, "share" shouldn't be codefonted, _but_ when you first mention "share extension" I would like to see a quick note or parenthetical that says what that is. Something simple and quick. I don't mind who lands/modifies this PR, whether it's @hellohuanlin or @antfitch, but I'd like this clarification.

Author: thanks for the comment. I think we should be fine here, as "share extension" is an official term by apple, so whoever wants to implement this feature should probably already know about it (also we have pictures right in this section so should be self-explanatory). It may be a good idea to link to apple's official doc tho.

            view.addSubview(flutterViewController.view)
            flutterViewController.view.frame = view.bounds
        }

        override func viewDidDisappear(_ animated: Bool) {

>             super.viewDidDisappear(animated)

Author: Some context on calling in `viewDidDisappear`: https://github.com/flutter/flutter/issues/165829#issuecomment-2787702265

            self.extensionContext?.cancelRequest(
                withError: NSError(domain: Bundle.main.bundleIdentifier!, code: 0))
        }
    }
    ```
:::note
1. If you do not cancel the request in `viewDidDisappear`, the `ShareViewController` will not be deallocated after being dismissed. 

2. You have to pass in the bundle identifier as the error domain, otherwise it will result in a `EXC_BAD_INSTRUCTION` crash. 
:::

## Advanced scrolling behavior

By default, flutter view does not receive any touch event in a share extension. 
To support a scrollable list in the share extension, follow [this instruction][]. 

> [this instruction]: {{site.github}}/flutter/website/issues/164670#issuecomment-2740702986

Author: I simply [linked the instruction](https://github.com/flutter/flutter/issues/164670#issuecomment-2761952022) instead of copying it, since it's not a very common usage, and I wanted to keep the instruction short. But I'm ok either way.

Reviewer: No, we don't want to link to instructions in a bug. Eventually the bug will be closed. Please put the information here. (I'd also like to edit it.)

Author: Oh I didn't do this yet, but we intend to simplify the API, so it's fine - let's keep it as it is for now.

> :::note
> 1. If you do not cancel the request in `viewDidDisappear`, the `ShareViewController` will not be deallocated after being dismissed. 
> 
> 2. You have to pass in the bundle identifier as the error domain, otherwise it will result in an `EXC_BAD_INSTRUCTION` crash. 
> :::
> 
> ## Advanced scrolling behavior
> 
> By default, flutter view does not receive any touch event in a share extension. 
> To support a scrollable list in the share extension, follow [this instruction][]. 
> 
> [this instruction]: {{site.github}}/flutter/website/issues/164670#issuecomment-2740702986

Reviewer: ```suggestion
:::note
1. If you _don't_ cancel the request in `viewDidDisappear`,
   the `ShareViewController` won't be deallocated after being dismissed. 

2. You must pass in the bundle identifier as the error domain,
   otherwise it crashes with an `EXC_BAD_INSTRUCTION` error. 
:::

## Advanced scrolling behavior

By default, flutter view does not receive any touch event in a share extension. 
To support a scrollable list in the share extension, follow [this instruction][]. 

[this instruction]: {{site.github}}/flutter/website/issues/164670#issuecomment-2740702986
```

Author: Done
